pkg/appStore/deployment/tool/gitops/AppStoreDeploymentArgoCdService.go

```diff
@@ -419,14 +420,17 @@ func (impl AppStoreDeploymentArgoCdServiceImpl) GetDeploymentHistory(ctx context
 			return result, err
 		}
 		for _, updateHistory := range versionHistory {
-			emailId := "anonymous"
+			emailId := bean.ANONYMOUS_EMAIL_ID
 			user, err := impl.userService.GetByIdIncludeDeleted(updateHistory.CreatedBy)
 			if err != nil && !util.IsErrNoRows(err) {
 				impl.Logger.Errorw("error while fetching user Details", "error", err)
 				return result, err
 			}
 			if user != nil {
 				emailId = user.EmailId
+				if !user.Exist {
+					emailId = fmt.Sprintf("%s (inactive)", user.EmailId)
```

Comment:
don't use user repository directly, wrap this logic in user service and use user service

Reply:
fixed 👍